### DISCUSSION TITLE
Revalidate page knowing that it will give 404

### DISCUSSION THREAD

**Discussion by Noitidart** [vercel/next.js#39439] [Help] [answered]
On deletion of an event (the entity) I want to revalidate the `/events/[eventId]` page. Calling revalidate on this is causing it to show the "event not found" 404 page, this is good otherwise it would show the event that's no longer around. However revalidate throws error when 404 status is returned. I'm using `res.unstable_revalidate` in Next.js 12.1 - the error thrown is not helpful, is there a way to catch the error to see what it is, that way I can choose to ignore the error, or succeed.

**Comment by ijjk** [comment, marked as answer]
Hi, in the latest version of Next.js `v12.2.4` this should no longer throw if `notFound: true` was returned and it will be able to detect this as a successful revalidation. Also note `res.unstable_revalidate` has been renamed to `res.revalidate` as it is now stable.

**Reply by Noitidart** [reply]
Oh thank you! I will test this out!
